app/upgrades/v16/upgrades.go

```diff
@@ -26,7 +70,76 @@ func CreateUpgradeHandler(
 		// Enable ICA controller
 		keepers.ICAControllerKeeper.SetParams(ctx, icacontrollertypes.DefaultParams())
 
+		// Add initial rate limits
+		if err := AddRateLimits(ctx, keepers.RatelimitKeeper); err != nil {
+			return vm, err
+		}
+
 		ctx.Logger().Info("Upgrade complete")
 		return vm, err
 	}
 }
+
+// Add rate limits as per https://www.mintscan.io/cosmos/proposals/890
+func AddRateLimits(ctx sdk.Context, k ratelimitkeeper.Keeper) error {
+	ctx.Logger().Info("Adding rate limits...")
+
+	// Osmosis
+	msg := RateLimits["osmosis-1"]
```

Comment:
Why don't we have a `for` loop over `RateLimits` to add all rate limits at once? If ordering is of importance, I guess we could make `RateLimits` a `slice` and not a `map`?

Reply:
Sure, but given that the code is already written, I don't see any issues with it.